### DISCUSSION TITLE
Edgemesh on which nodes?

### DISCUSSION THREAD

**Discussion by alb12393** [kubeedge/kubeedge#4948] [Q&A]
Using EdgeMesh to access applications on the edge since Kube-flannel does not seem to be supported (Kube-flannel is on every other node). I initially assumed that EdgeMesh would only be necessary on the edge node but it seems to only work once EdgeMesh is present on every node (i.e. completely replacing Kube-flannel). Is this the intended behavior or am I doing something wrong? 

FYI by EdgeMesh working/not working I'm specifically referring to the test cases listed in the documentation: https://edgemesh.netlify.app/guide/test-case.html#prepare.

Client Version: v1.27.3, KubeEdge v1.12.1, EdgeMesh v1.14.0

**Comment by ravaga** [comment]
You don't have to install EdgeMesh in all Edge nodes, but you have to install it only in the nodes in which you want to achieve network cross-comunication and DNS resolution (you need EdgeMesh to be able to call K8s service names in the pods). In addition, you don't need to install a third-party CNI plugin (e.g. Flannel or Cilium) in Edge nodes, just write a simple CNI configuration using the _bridge_ CNI plugin, for instance.

For more information, please check the answer I wrote in this discussion #5145
